contracts/DXswapFeeReceiver.sol

```diff
@@ -6,122 +6,226 @@ import './interfaces/IWETH.sol';
 import './libraries/TransferHelper.sol';
```

Comment:
I'd honestly upgrade to Solidity v0.8.13 or later if available and leverage both:

1. The fact that with Sol >=v0.8.0, safe math is not really required anymore since under/overflow checks are automatically performed.
2. The Yul optimizer is production ready and I'd use it to shave off some gas consumption.